bin/bs_ios

```diff
@@ -0,0 +1,144 @@
+	echo "BROWSERSTACK_IOS_DEVICES not set, using default: $default_devices"
+	BROWSERSTACK_IOS_DEVICES="$default_devices"
+fi
+
+if [ "$BROWSERSTACK_SKIP_BUILD" = 1 ]; then
+	echo "BROWSERSTACK_SKIP_BUILD set to 1, build was skipped"
+else
+	patrol build ios --release "${original_args[@]}"
+fi
+
+echo "Will create zip archive of test files"
+
+cd build/ios_integ/Build/Products
+
+rm -rf Payload && mkdir -p Payload
+cp -r "Release${FLAVOR_SUFFIXED}iphoneos/Runner.app" Payload
+zip -r Runner.ipa Payload >/dev/null
+
+cd - >/dev/null
+
+cd "build/ios_integ/Build/Products/Release${FLAVOR_SUFFIXED}iphoneos"
+rm -rf ios_tests.zip
+
+# BrowserStack fails if DiagnosticCollectionPolicy is present
+plutil -remove 'TestConfigurations.TestTargets.DiagnosticCollectionPolicy' ../"${FLAVOR_PREFIXED}"_TestPlan_iphoneos*.xctestrun
+
+cp ../"${FLAVOR_PREFIXED}"_TestPlan_iphoneos*.xctestrun .
+zip -r ios_tests.zip "${FLAVOR_PREFIXED}"_TestPlan_iphoneos*.xctestrun RunnerUITests-Runner.app >/dev/null
+cd - >/dev/null
+
+echo "Created zip archive"
+
+app_path="$PWD/build/ios_integ/Build/Products/Runner.ipa"
+if [ ! -f "$app_path" ]; then
+	echo "Error: app under test not found at $app_path"
+	exit 1
+fi
+
+# https://www.browserstack.com/docs/app-automate/api-reference/xcuitest/apps#upload-an-app
+printf "Will upload app under test from %s\n\n" "$app_path"
+app_upload_response="$(
+	curl -u "$BROWSERSTACK_CREDS" \
+		-X POST "https://api-cloud.browserstack.com/app-automate/xcuitest/v2/app" \
+		-F "file=@$app_path"
+)"
+
+app_url="$(echo "$app_upload_response" | jq --raw-output .app_url)"
+printf "\nUploaded app under test, url: %s\n" "$app_url"
+
+test_path="$PWD/build/ios_integ/Build/Products/Release${FLAVOR_SUFFIXED}iphoneos/ios_tests.zip"
+if [ ! -f "$test_path" ]; then
+	echo "Error: zip archive of test suite not found at $test_path"
+	exit 1
+fi
+
+# https://www.browserstack.com/docs/app-automate/api-reference/xcuitest/tests#upload-a-test-suite
+printf "Will upload zip archive of test suite from %s\n\n" "$test_path"
+test_upload_response="$(
+	curl -u "$BROWSERSTACK_CREDS" \
+		-X POST "https://api-cloud.browserstack.com/app-automate/xcuitest/v2/test-suite" \
+		-F "file=@$test_path"
+)"
+
+test_url="$(echo "$test_upload_response" | jq --raw-output .test_suite_url)"
+printf "\nUploaded zip archive of test suite, url: %s\n" "$test_url"
+
+# https://www.browserstack.com/docs/app-automate/api-reference/xcuitest/builds#execute-a-build
+printf "Will schedule test execution\n\n"
+curl -u "$BROWSERSTACK_CREDS" \
+	-X POST "https://api-cloud.browserstack.com/app-automate/xcuitest/v2/xctestrun-build" \
+	-H "Content-Type: application/json" \
+	--data-binary @- <<EOF
+{
+    "app": "$app_url",
+    "testSuite": "$test_url",
+    "project": "$BROWSERSTACK_PROJECT",
+    "devices": $BROWSERSTACK_IOS_DEVICES,
+    "deviceLogs": true,
+    "enableResultBundle": true,
+    "idleTimeout": $BROWSERSTACK_IDLE_TIMEOUT
+}
+EOF
+
+printf "\n\nScheduled test execution\n"
```

Comment:
if curl response is not 200 `printf "\n\nScheduled test execution\n"` still will be executed

Reply:
very good point. I think we can use [curl's `--fail` flag](https://superuser.com/a/657174/721371) for that and then just check if the exit code of last executed command is not equal to `0`:

```
curl --fail <... >
if [ $? != 0 ]; then
  echo "curl failed"
fi
```

Reply:
Actually, we don't even need to check exit code, because we do `set -e` at the top of the script.